### DISCUSSION TITLE
how to track button presses without ending trial? / how not to end trial on certain button presses?

### DISCUSSION THREAD

**Discussion by neuronflow** [jspsych/jsPsych#673] [Q&A]
I am running an experiment using html-keyboard-responses:
https://www.jspsych.org/plugins/jspsych-html-keyboard-response/

The trial should be ended once users come up with a decision and press the according button. So far so good, but users can also press the space bar to display some hints. Pressing the space bar should not end the trial. I want to track how often and when users press the space bar.

I could set response_ends_trial to false. I think this should record all keypresses?
But then the actual responses need to end the trial.
What is the smartest way to achieve that?

**Comment by neuronflow** [comment]
I managed to implement most of this with a keyboard listener..I only need to find a way to attach the information to the trials now within the on_finish callback.

**Comment by jodeleeuw** [comment]
This sounds like a case where modifying the plugin file to accommodate the hints would be a good idea. Is that what you ended up doing?
